internal/p2p/peermanager.go

```diff
@@ -1366,48 +1366,45 @@ func (s *peerStore) Ranked() []*peerInfo {
 		s.ranked = append(s.ranked, peer)
 	}
 	sort.Slice(s.ranked, func(i, j int) bool {
-		return s.ranked[i].Score() > s.ranked[j].Score()
-		// TODO: reevaluate more wholistic sorting, perhaps as follows:
-
-		// // sort inactive peers after active peers
-		// if s.ranked[i].Inactive && !s.ranked[j].Inactive {
-		// 	return false
-		// } else if !s.ranked[i].Inactive && s.ranked[j].Inactive {
-		// 	return true
-		// }
-
-		// iLastDialed, iLastDialSuccess := s.ranked[i].LastDialed()
-		// jLastDialed, jLastDialSuccess := s.ranked[j].LastDialed()
-
-		// // sort peers who our most recent dialing attempt was
-		// // successful ahead of peers with recent dialing
-		// // failures
-		// switch {
-		// case iLastDialSuccess && jLastDialSuccess:
-		// 	// if both peers were (are?) successfully
-		// 	// connected, convey their score, but give the
-		// 	// one we dialed successfully most recently a bonus
-
-		// 	iScore := s.ranked[i].Score()
-		// 	jScore := s.ranked[j].Score()
-		// 	if jLastDialed.Before(iLastDialed) {
-		// 		jScore++
-		// 	} else {
-		// 		iScore++
-		// 	}
-
-		// 	return iScore > jScore
-		// case iLastDialSuccess:
-		// 	return true
-		// case jLastDialSuccess:
-		// 	return false
-		// default:
-		// 	// if both peers were not successful in their
-		// 	// most recent dialing attempt, fall back to
-		// 	// peer score.
-
-		// 	return s.ranked[i].Score() > s.ranked[j].Score()
-		// }
+		// sort inactive peers after active peers
+		if s.ranked[i].Inactive && !s.ranked[j].Inactive {
+			return false
+		} else if !s.ranked[i].Inactive && s.ranked[j].Inactive {
+			return true
+		}
+
+		iLastDialed, iLastDialSuccess := s.ranked[i].LastDialed()
+		jLastDialed, jLastDialSuccess := s.ranked[j].LastDialed()
+
+		// sort peers who our most recent dialing attempt was
+		// successful ahead of peers with recent dialing
+		// failures
+		switch {
+		case iLastDialSuccess && jLastDialSuccess:
+			// if both peers were (are?) successfully
+			// connected, convey their score, but give the
+			// one we dialed successfully most recently a bonus
+
+			iScore := int64(s.ranked[i].Score())
+			jScore := int64(s.ranked[j].Score())
+			if jLastDialed.Before(iLastDialed) {
+				jScore++
+			} else {
+				iScore++
+			}
+
+			return iScore > jScore
+		case iLastDialSuccess:
+			return true
+		case jLastDialSuccess:
+			return false
+		default:
+			// if both peers were not successful in their
+			// most recent dialing attempt, fall back to
+			// peer score.
+
```

Comment:
I think this part overly complicates things. Peers who we failed to dial will already have a lower score than those that were successful.

Reply:
right. I think it was the case that peers in the list would often cluster at specific scores, so providing an extra oppertunity to tiebreak peers would be useful. Maybe by expanding the range some this becomes less of an issue. 

At the same time, we only decrement scores for failed dialing attempts, so this would give us more recent data, and allows us to bubble up nodes that have succeeded more recently, or at least that's the hope.